cw-rpm-utils

```diff
@@ -0,0 +1,75 @@
+#!/bin/bash
+#
+# Utilities for building RPM packages
+#
+# Note that this script is only ever expected to be source-d into another
+# script - the !/bin/bash comment above is just so that vim can do
+# syntax-highlighting correctly.
+#
+# Project Clearwater - IMS in the Cloud
+# Copyright (C) 2016  Metaswitch Networks Ltd
+#
+# This program is free software: you can redistribute it and/or modify it
+# under the terms of the GNU General Public License as published by the
+# Free Software Foundation, either version 3 of the License, or (at your
+# option) any later version, along with the "Special Exception" for use of
+# the program along with SSL, set forth below. This program is distributed
+# in the hope that it will be useful, but WITHOUT ANY WARRANTY;
+# without even the implied warranty of MERCHANTABILITY or FITNESS FOR
+# A PARTICULAR PURPOSE.  See the GNU General Public License for more
+# details. You should have received a copy of the GNU General Public
+# License along with this program.  If not, see
+# <http://www.gnu.org/licenses/>.
+#
+# The author can be reached by email at clearwater@metaswitch.com or by
+# post at Metaswitch Networks Ltd, 100 Church St, Enfield EN2 6BQ, UK
+#
+# Special Exception
+# Metaswitch Networks Ltd  grants you permission to copy, modify,
+# propagate, and distribute a work formed by combining OpenSSL with The
+# Software, or a work derivative of such a combination, even if such
+# copying, modification, propagation, or distribution would otherwise
+# violate the terms of the GPL. You must comply with the GPL in all
+# respects for all of the code used other than OpenSSL.
+# "OpenSSL" means OpenSSL toolkit software distributed by the OpenSSL
+# Project and licensed under the OpenSSL Licenses, or a work based on such
+# software and licensed under the OpenSSL Licenses.
+# "OpenSSL Licenses" means the OpenSSL License and Original SSLeay License
+# under which the OpenSSL Project distributes the OpenSSL toolkit software,
+# as those licenses appear in the file LICENSE-OPENSSL.
+
+export CW_RPM_UTILS_PACKAGE=$1
+export CW_RPM_UTILS_TOPDIR=$2
+export CW_RPM_UTILS_BUILDROOT=$3
+
+function setup_buildroot {
+  rm -rf ${CW_RPM_UTILS_BUILDROOT}
+  copy_to_buildroot build-infra/LICENSE /usr/share/licenses/$CW_RPM_UTILS_PACKAGE
+}
+
+function copy_to_buildroot {
+  mkdir -p ${CW_RPM_UTILS_BUILDROOT}/$2
+  cp -rp ${CW_RPM_UTILS_TOPDIR}/$1 ${CW_RPM_UTILS_BUILDROOT}/$2/$3
+}
+
+function install_to_buildroot {
+  export -f copy_to_buildroot
+  xargs -r -I {} bash -c 'copy_to_buildroot "$(cut -d\  -f 1 <<< "{}")" "$(cut -d\  -f 2 <<< "{}")"'
+}
+
+function dirs_to_buildroot {
+  xargs -r -I {} mkdir -p ${CW_RPM_UTILS_BUILDROOT}/{}
+}
+
+function build_files_list {
+  cd ${CW_RPM_UTILS_BUILDROOT}
+  find . -type f -a -executable |
+  sed -e 's/^\./%attr(755, -, -) /g' |
+  sed -e 's/ \(\/etc\/\)/ %config \1/g'
+  find . -type f -a ! -executable -o -type d -empty |
```

Comment:
Could you put a blank line here - it's easy to miss the lack of trailing pipe and read it as a continuation of the previous command.